packages/transaction-controller/src/TransactionController.ts

```diff
@@ -898,6 +902,50 @@ export class TransactionController extends BaseController<
     this.incomingTransactionHelper.stop();
   }
 
+  /**
+   * Adds external provided transaction to state as confirmed transaction.
+   *
+   * @param transactionMeta - TransactionMeta to add transactions.
+   * @param transactionReceipt - TransactionReceipt of the external transaction.
+   * @param baseFeePerGas - Base fee per gas of the external transaction.
+   */
+  async confirmExternalTransaction(
+    transactionMeta: TransactionMeta,
+    transactionReceipt: TransactionReceipt,
+    baseFeePerGas: Hex,
+  ) {
+    // Create logger
+    const log = createModuleLogger(projectLogger, 'confirmExternalTransaction');
+
+    // Run validation and add external transaction to state.
+    this.addExternalTransaction(transactionMeta);
```

Comment:
Given it's one extra line, shall we include this in the try catch also?

Reply:
Wouldn't it make the API validations silently fail since we throw there?

Reply:
Excellent point, ignore me 😄